protocol_grpc.go

```diff
@@ -846,6 +846,10 @@ func grpcContentTypeFromCodecName(web bool, name string) string {
 	if web {
 		return grpcWebContentTypePrefix + name
 	}
+	if name == codecNameProto {
+		// for compatibility with some servers prefer an implicit default codec
```

Comment:
Can we clarify _which_ servers we're talking about? It'd be nice to make it very explicit that we're talking about GCP's own frontends.